pkg/performanceprofile/controller/performanceprofile_controller.go

```diff
@@ -683,6 +455,34 @@ func hasFinalizer(profile *performancev2.PerformanceProfile, finalizer string) b
 	return false
 }
 
+func (r *PerformanceProfileReconciler) getAndValidateMCP(ctx context.Context, instance client.Object) (*mcov1.MachineConfigPool, *reconcile.Result, error) {
```

Comment:
can we push this function (or an amended form of) inside `err = r.StatusWriter.UpdateOwnedConditions(ctx, instance, profileMCP)`, so we can remove the potentially problematic `profileMCP` argument?

Reply:
Sure we do, sounds like a good idea

Reply:
I removed the `ProfileMCP` from the `UpdateOwnedConditions` as requested. Good idea, thank you